spec/classes/init_spec.rb

```diff
@@ -1,147 +1,177 @@
+          SELINUX=disabled
+          # SELINUXTYPE= type of policy in use. Possible values are:
+          # targeted - Only targeted network daemons are protected.
+          # strict - Full SELinux protection.
+          SELINUXTYPE=targeted
+          EOF
+          ) }
+        it { is_expected.to contain_package('mcstrans').with(ensure: 'installed') }
 
-          if os_facts[:os][:release][:major].to_i >= 7
-            it { is_expected.to create_service('mcstransd').with({
-              :enable => true,
-              :ensure => 'stopped'
-            }) }
-            it { is_expected.to contain_package('policycoreutils-restorecond').with(:ensure => 'installed') }
-          else
-            it { is_expected.to create_service('mcstrans').with({
-              :enable => true,
-              :ensure => 'stopped'
-            }) }
-            it { is_expected.to contain_package('policycoreutils').with(:ensure => 'installed') }
-          end
-          it { is_expected.to create_service('restorecond').with({
-            :enable => true,
-            :ensure => 'stopped'
-          }) }
+        if os_facts[:os][:release][:major].to_i >= 7
+          it { is_expected.to create_service('mcstransd').with(
+            enable: true,
+            ensure: 'stopped'
+          ) }
+          it { is_expected.to contain_package('policycoreutils-restorecond').with(ensure: 'installed') }
+        else
+          it { is_expected.to create_service('mcstrans').with(
+            enable: true,
+            ensure: 'stopped'
+          ) }
+          it { is_expected.to contain_package('policycoreutils').with(ensure: 'installed') }
         end
+        it { is_expected.to create_service('restorecond').with(
+          enable: true,
+          ensure: 'stopped'
+        ) }
+      end
+
+      context 'with mode set' do
+        let(:params) {{ mode: 'mls' }}
+        it { is_expected.to compile.with_all_deps }
+        it { is_expected.to create_file('/etc/selinux/config').with_content(<<-EOF.gsub(/^\s+/,'')
+          # This file controls the state of SELinux on the system.
+          # SELINUX= can take one of these three values:
+          # enforcing - SELinux security policy is enforced.
+          # permissive - SELinux prints warnings instead of enforcing.
+          # disabled - SELinux is fully disabled.
+          SELINUX=enforcing
+          # SELINUXTYPE= type of policy in use. Possible values are:
+          # targeted - Only targeted network daemons are protected.
+          # strict - Full SELinux protection.
+          SELINUXTYPE=mls
+          EOF
+          ) }
+      end
+
+      context 'with manage_utils_package => false' do
+        let(:params) {{ manage_utils_package: false }}
+        it { is_expected.to_not contain_package('policycoreutils-python') }
+      end
 
-        context 'with mode set' do
-          let(:params) {{ :mode => 'mls' }}
+      context 'notifying user of reboots' do
```

Comment:
To be complete, it would be best to test transitions with ensure values of true, false, and 'permissive'.